### ISSUE TITLE
 The redundantReturn rule is unexpectedly being applied below "// swiftformat:disable:next redundantReturn"

### ISSUE THREAD

**Issue by acosmicflamingo** [nicklockwood/SwiftFormat#1509]
With the new SwiftFormat 0.52.1 release (was working in 0.52.0), I am noticing that applying `swiftformat` results in removing `return` from a line where the following precedes it: `// swiftformat:disable:next redundantReturn`. Has this rule been renamed?

Here's an example:
```diff
public func UIConfirmationDialogController<Action: Equatable>(
  state: UIAlertState<Action>,
  send: @escaping (Action?) -> Void
) -> UIAlertController {
  // swiftformat:disable:next redundantReturn
- return UIAlertController(state: state, send: send, preferredStyle: .actionSheet)
+ UIAlertController(state: state, send: send, preferredStyle: .actionSheet)
}
```

**Comment by nicklockwood** [comment]
@acosmicflamingo this seems to be a new bug. The rule name hasn't changed.

**Comment by nicklockwood** [comment]
@acosmicflamingo something like this should still work:

```swift
// swiftformat:disable redundantReturn
public func UIConfirmationDialogController<Action: Equatable>(
  state: UIAlertState<Action>,
  send: @escaping (Action?) -> Void
) -> UIAlertController {
  return UIAlertController(state: state, send: send, preferredStyle: .actionSheet)
}
// swiftformat:enable redundantReturn
```

**Comment by acosmicflamingo** [comment]
Fascinating. Your code snippet does work, but this does not:
```swift
public func UIConfirmationDialogController<Action: Equatable>(
  state: UIAlertState<Action>,
  send: @escaping (Action?) -> Void
) -> UIAlertController {
  // swiftformat:disable redundantReturn
  return UIAlertController(state: state, send: send, preferredStyle: .actionSheet)
  // swiftformat:enable redundantReturn
}
```

Is there anything you want me to run on my end that can help with debugging?

**Comment by nicklockwood** [comment]
@acosmicflamingo I just fixed it :-)

**Comment by acosmicflamingo** [comment]
@nicklockwood yay! Thanks a bunch :D

**Comment by nicklockwood** [comment]
@acosmicflamingo fix landed in 0.52.2
